fixieai/cli/agent_config.py

```diff
@@ -0,0 +1,135 @@
+import dataclasses
+import os
+from typing import Callable, Optional, TextIO, Union
+
+import dataclasses_json
+import prompt_toolkit
+import yaml
+
+from fixieai.cli import validators
+
+
+@dataclasses.dataclass
+class AgentConfig(dataclasses_json.DataClassJsonMixin):
+    """Represents an agent.yaml config file."""
+
+    agent_id: str = "agent_id"
+    name: str = ""
+    description: str = ""
+    entry_point: str = "main.py"
+    more_info_url: str = ""
+    deployment_url: str = ""
+    public: bool = False
+
+    @classmethod
+    def from_yaml(cls, config: Union[str, TextIO]) -> "AgentConfig":
+        """Loads AgentConfig from stream or string of yaml config."""
+        return AgentConfig.from_dict(yaml.safe_load(config))
+
+    def to_yaml(self) -> str:
+        """Dumps AgentConfig as a yaml config."""
+        return yaml.dump(self.to_dict(), sort_keys=False)  # type: ignore
+
+
+def load_config(path: Optional[str] = None) -> "AgentConfig":
+    """Loads AgentConfig from the given path, or its default path.
+
+    Args:
+        path: Optional path to load config from. By default, config is loaded from
+            "{cwd}/agent.yaml".
+    """
+    if path is None:
+        path = os.path.join(os.getcwd(), "agent.yaml")
+    with open(path, "r") as fp:
+        return AgentConfig.from_yaml(fp)
+
+
+def save_config(agent_config: AgentConfig, path: Optional[str] = None):
+    """Saves the given AgentConfig to the given path, or its default path.
+
+    Args:
+        agent_config: The AgentConfig object to save into file.
+        path: Optional path to save. By default, it's saved to the default path at
+            "{cwd}/agent.yaml".
+    """
+    if path is None:
+        path = os.path.join(os.getcwd(), "agent.yaml")
+    with open(path, "w") as fp:
+        fp.write(agent_config.to_yaml())
+
+
+def prompt_user(agent_config: AgentConfig):
+    """Prompts the to edit values in AgentConfig."""
+    print(
+        """This utility will walk you through creating an agent.yaml file.
+    
+"""
+    )
+    agent_config.agent_id = _prompt_user_for_str(
+        "agent id", agent_config.agent_id, validators.validate_agent_id
+    )
+    agent_config.description = _prompt_user_for_str(
+        "description",
+        agent_config.description,
+    )
+    agent_config.entry_point = _prompt_user_for_str(
+        "entry point",
+        agent_config.entry_point,
+    )
+    agent_config.more_info_url = _prompt_user_for_str(
+        "more info url",
+        agent_config.more_info_url,
+        validators.validate_url,
+    )
+    agent_config.public = _prompt_user_for_bool(
+        "public?",
+        agent_config.public,
+    )
+
+
+def _prompt_user_for_str(
```

Comment:
I believe this code is unnecessary if you use Click's option prompting support.

https://click.palletsprojects.com/en/8.1.x/options/#prompting

The idea here being that the options should be specified on the command line, using flags, but if the flags are not provided then Click prompts the user for them. The resulting value is then passed down as an option in the same way as if it were specified on the command line. This is more consistent and saves code.

Reply:
Great suggestion. We are using more of click now and I think it's in a much better shape.